src/inc/Lfo.h

```diff
@@ -0,0 +1,119 @@
+#ifndef __Lfo_Hdr__
+#define __Lfo_Hdr__
+
+#include "RingBuffer.h"
+#include <cmath>
+
+class CWavetableLFO
+{
+public:
+    /*! Creates intance for the wavetable LFO
+    \param iWavetableLength wavetable length of the LFO
+    \param iSampleRate sample rate for the LFO
+    \param fFrequency frequency for the LFO in Hz
+    \param fAmplitude amplitude for the LFO
+    */
+    CWavetableLFO(int iWavetableLength, int iSampleRate, float fFrequency, float fAmplitude):
+        m_iSampleRate(iSampleRate),
+        m_dReadIdx(0)
+    {
+        double const l_dPi = 3.14159265358979323846;
+        assert(iWavetableLength > 0);
+        pCWavetableRingBuff = new CRingBuffer<float>(iWavetableLength);
+        for (int i = 0; i < iWavetableLength; i++)
+        {
+            pCWavetableRingBuff->putPostInc(sin(2 * l_dPi * i / iWavetableLength));
+        }
+        m_fAmplitude = fAmplitude; 
+        m_dReadInc = static_cast<double>(pCWavetableRingBuff->getLength()) / static_cast<double>(m_iSampleRate) * fFrequency;
+    }
+
+    ~CWavetableLFO()
+    {
+        delete pCWavetableRingBuff;
+        pCWavetableRingBuff = 0;
+    }
+
+    /*! Set the phase for the read head
+    \param fPhaseInDegrees phase for the read head in degrees
+    */
+    void setReadPhase(float fPhaseInDegrees)
+    {
+        while (fPhaseInDegrees >= 360)
+            fPhaseInDegrees -= 360;
+        while (fPhaseInDegrees < 360)
```

Comment:
Yes it should be. My mistake here, will fix it later. However it won't cause an error since this would be automatically wrapped back in the ringbuffer get function.